### PR TITLE
Implement Event observer and Settings for 3rd party dep injection

### PR DESCRIPTION
**Pull request by frrist**
### What
- Implement TipSetObserver interface for observing tipset events (apply/revert)
- Implement Settings opts for 3rd party dep injection (allows Visor to instantiate a Lotus node)

### CODE REVIEW

File: Makefile
Comment (magik6k) [inline Makefile:231]:
```suggestion
.PHONY: lotus-sentinel
```

File: cmd/lotus-sentinel/daemon.go
Comment (magik6k) [inline cmd/lotus-sentinel/daemon.go:143]: I'd try to deduplicate this with the 'normal' daemon

File: node/builder.go
Comment (magik6k) [inline node/builder.go:17]:
```suggestion
	sentinelapi "github.com/filecoin-project/lotus/cmd/lotus-sentinel/api"
```

**Review by magik6k — COMMENTED** [review]
First pass - looks pretty good

(I'm assuming you want a review from @iand before merging)

File: cmd/lotus-sentinel/daemon.go
Comment (frrist) [reply cmd/lotus-sentinel/daemon.go:143]: Any preference for where this should live? In order for other packages to use it, we'll have to move it out of this file since it's in `package main` -- my first thought was the repo package.

Comment (iand) [reply cmd/lotus-sentinel/daemon.go:143]: Would prefer not to have to make the user switch between binaries for this task. Let's find a new home for this function in a suitable lotus package.

**Review by iand — COMMENTED** [review]
Pretty happy with how this turned out. Now the question is whether the binary needs to live in the lotus repo or can it live in sentinel's? The extension points and exposure of things like ImportChain obviously have to stay in Lotus but once we have those it seems by moving the binary elsewhere we could avoid Lotus depending on the all the visor and postgres packages. Plus we get to decouple the development processes of the two projects.

**Comment by frrist** [comment]
I'm going to extract the lotus-sentinel binary code to the visor repo and modify this PR only to contain the changes required to make that work -- I suspect they will be minimal (sorry for the churn)

**Comment by frrist** [comment]
Converting to a draft as this will require more changes than I anticipated. The other half lives here: https://github.com/filecoin-project/sentinel-visor/pull/408

**Review by magik6k — APPROVED** [review]

**Pull request merged**
